hbase-common/src/main/java/org/apache/hadoop/hbase/HConstants.java

```diff
@@ -136,6 +136,12 @@
   /** Default value for the max percent of regions in transition */
   public static final double DEFAULT_HBASE_MASTER_BALANCER_MAX_RIT_PERCENT = 1.0;
 
+  /** Time in milliseconds to wait meta region assignment, when moving non-meta regions. */
+  public static final String HBASE_MASTER_WAITING_META_ASSIGNMENT_TIMEOUT =
+    "hbase.master.waiting.meta.assignment.timeout";
+
+  public static final long HBASE_MASTER_WAITING_META_ASSIGNMENT_TIMEOUT_DEFAULT = 10000;
```

Comment:
I think it's fine to keep 5 sec, my only concern was that sometimes when the system is overall slow, meta might take few more seconds (which is not good for the cluster's availability though) and hence before failing this transaction fast, I thought maybe we can wait for 10s. However this is subjective to usecase and interpretation of individual users so I think any value is fine here. We are anyways making this configurable for that reason only, so let's keep 10 sec maybe and user can configure based on their usecase.

Reply:
Clear.